### PR TITLE
[Redesign] User authentication pages and base structure for further changes

### PR DESCRIPTION
**Pull request by fernandoabolafio**
This PR deals with the first milestone of the [new design implementation](https://github.com/decred/politeiagui/issues/1205) which is:

- User authentication: Login, Signup, Reset Password, Verify account email, Privacy Policy, e2e integrations. 
The checklist of this PR can be found in the issue https://github.com/decred/politeiagui/issues/1205, where each milestone is better described.

I tried to keep the changes for the new version isolated as possible from the current application so we don't have major conflicts and functionalities regressions. To accomplish that the `src/index.js` is loading a different 'App' for each case. I used the "v2" suffix in the related folder/files in order to differentiate what is being used only in the redesign process.

- To start version 2 run: `yarn start-v2`
- To build version 2 run: `yarn build-v2`

The preview of the current state of the application can be found here: https://pi-redesign.authit.co. Please, refer only to the pages related to the "User authentication" context as they are the ones which are expected to be fully complete and functional in this PR. Other pages, such as Public Proposals or Create Proposal, are there just to give a better sense of the website overall structure.

### CODE REVIEW

**Comment by MariaPleshkova** [comment]
Well. As we're still working on redesigning Pi, there were some small tweaks made on these screens.
image
"Reset password" was moved left. And the same was made for Sign up ("Resend verification email" was moved left too).
image

**Now regarding more specific things:**

**1.** Log in/Create a new account titles need to be Regular font, not Semi-bold.
Also their size should be 28px.

**2.** The size of the normal text should be 16px (Email, password, etc.)
image

**3.** We also need to add hover effects on all clickable elements (blue/gray links). I suggest to change them to Semi-bolds.
image

**4.** Change the color of “Privacy policy” sign to _#8997A5_

**5.** Add hover effect to non-active header item (Log in/Sign up). Hover effect - _#091440_

**6.** Main white section (rectangle) should be on the center of the screen (up/down and left/right)
image

**7.** Notification text should starts with the capital letter. Its size has to be 13px.
The size of fields titles (that were moved up) - 13px as well.
image

**8.** Not sure if Privacy Policy screen is already finished, but just in case:
It has to be a pop up.
image

Basically, amends for the rest created screens are the same as for Log in screen above (fonts sizes/weights, color, etc.)

So these are the ones I've noticed.

**Comment by lukebp** [comment]
One thing I noticed is that the "required" text causes the form entry fields below it to be pushed down the page.

**Comment by fernandoabolafio** [comment]
@MariaPleshkova thanks for your comments, they are all pretty straight forward to get fixed. I have comments about #6 and #8:
- About #6 (Main white section (rectangle) should be on the center): I first placed the main area at the center of the page. For login it and other smaller contents such as "Resend verification email" it looks nice. However, for the Signup form, for instance, it causes the page content to overflow the viewport, thus requiring the user to scroll the page to fill up the form. This can be avoided just by having the Form placed closer to the top of the page. What do you think?
- About #8 (Not sure if Privacy Policy screen is already finished...): I chose to go with a separated page for privacy policy because of this: https://github.com/decred/politeiagui/issues/771. We recently decided to move the Privacy Policy from a pop up to a separated page, so I figure that would be better to stick with that.

**Comment by MariaPleshkova** [comment]
@fernandoabolafio 

- About 6: so do you suggest to keep only long forms closer to the header? Or to leave all of them as you made at first?

- About 8: hm, but in this case we have to add at least Back button there. As when user opened Privacy policy and then he wants to go back, he needs to click on Log in/Sign up again. Also what will happen if user already filled the fields and then decided to click on Privacy Policy? Will he lose this entered data?
IMHO there are only 1-2 paragraphs there, so not sure that we need a separate page for this. 
@linnutee what do you think?

Regarding an updated link, I'll send this asap. Finishing with some last tweaks.

**Comment by fernandoabolafio** [comment]
@MariaPleshkova thanks for the quick reply. 
- About 6: I think all forms for the Authentication pages (Signup, login, resend email etc) could be closer to the top. IMO it's easier to read and we keep the form position consistent across the different pages.
- About 8: I don't have strong feelings about this. I am okay with both approaches. However, I tend to like the modal better for the reasons you mentioned above.

**Comment by linnutee** [comment]
Regarding 8, i'd say modal is more fitting as there's quite little content. If the privacy policy happens to increase in content size in the future or become more significant then there's some more reason to make a subpage for it. 

I'll also mention regarding graphic assets, logo for example – wherever possible use svg's for icons or any vector based assets.  Then there's less play around the @2x and @3x png's.

**Comment by tiagoalvesdulce** [comment]
@linnutee @MariaPleshkova we are building a UI lib with live documentation ([pi-ui](https://github.com/tiagoalvesdulce/pi-ui)). Can you guys take a look and write issues there? I think it's more productive that way. Also, we are missing the UI example of some components like the Dropdown in the original Adobe XD document.

**Comment by fernandoabolafio** [comment]
@linnutee @MariaPleshkova I addressed your main concerns regarding the font-size and the privacy policy. Now the privacy policy is displayed in a modal but there is also a "permalink" which redirects to the full route in case we need to link from other places.
Screen Shot 2019-06-11 at 21 43 29

It is missing the "i" icon from the specs because we need to add that to the pi-ui lib yet. I will make a point to address those minor details later if that is okay for you.

**Comment by lukebp** [comment]
A few comments.  The first two are directed towards @linnutee @MariaPleshkova.

1. Anytime a form field fails the validation, error text such as "Required" or "Invalid Email" pops up.  When the error text appears, all of the elements below it get shifted down.  Is this intentional?

2. This is the text that appears on the screen when you sign up.

> Politeia will send you a link to verify your email address. You must open this link in the same browser. After verifying your email, Politeia will create your Politeia “identity”, which consists of a public/private cryptographic key pair and browser cookie. This is necessary to verify your identity and allow submission of proposals, commenting, voting, and other Politeia functions. After completing the signup process, you can export your identity (public/private keys) to another browser at any time.

tbh I didn't read any of it.  I just pushed the "I understand, sign me up" button.  A big block of text like 
that is pretty unappealing and I'm not sure very many people will actually read it.  My suggestion would be to keep it short and to the point:

"Politeia will send you a link to verify your email address. You must open this link in the same browser." 

And then save the user identity explanation for the user onboarding modal that will appear the first time the user logs in.

3. @fernandoabolafio: is the "Verify Email    [dev]" element suppose to be there?
Screen Shot 2019-06-11 at 4 40 11 PM

**Comment by fernandoabolafio** [comment]
@lukebp I will answer you here again just for documentation purposes. Yes, this [dev] tag is intentional to wrap actions which will no be available in a production website. This is just to simplify the process of going through the screens without leaving the website.

File: src/componentsv2/Sidebar.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Sidebar as UISidebar, Button, Card } from "pi-ui";
+import { withRouter } from "react-router-dom";
+import LoggedInContent from "./LoggedInContent";
+import StaticMarkdow from "./StaticMarkdown";
```
Comment (tiagoalvesdulce) [inline src/componentsv2/Sidebar.jsx]: There's a typo here

File: src/componentsv2/Sidebar.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        <Button
+          className="margin-bottom-s"
+          onClick={() => history.push(`/proposals/new`)}
+          fullWidth
+        >
+          Submit Proposal
+        </Button>
+      </LoggedInContent>
+      <Card paddingSize="small">
+        <StaticMarkdow contentName={aboutContent} />
```
Comment (tiagoalvesdulce) [inline src/componentsv2/Sidebar.jsx]: Same as above

File: src/containers/User/Login/Form.jsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  return (
+    <>
+      <FormWrapper
+        initialValues={{
+          email: "",
+          password: ""
+        }}
+        loading={!validationSchema}
+        validationSchema={validationSchema}
+        onSubmit={async (
```
Comment (tiagoalvesdulce) [inline src/containers/User/Login/Form.jsx]: I think these functions shouldn't be inline in the JSX, can you create a named function in the component instead?

File: src/containers/User/ResetPassword/RequestForm.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  const [email, setEmail] = useState(false);
+  const success = !!email;
+  return (
+    <>
+      <FormWrapper
+        initialValues={{
+          email: ""
+        }}
+        validationSchema={validationSchema}
+        onSubmit={async (values, { setSubmitting, setFieldError }) => {
```
Comment (tiagoalvesdulce) [inline src/containers/User/ResetPassword/RequestForm.jsx]: Same as above.

File: src/containers/User/ResetPassword/ResetForm.jsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  if (invalidParamsError) {
+    throw invalidParamsError;
+  }
+
+  return (
+    <FormWrapper
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      loading={!validationSchema}
+      onSubmit={async (values, { setSubmitting, setFieldError, resetForm }) => {
```
Comment (tiagoalvesdulce) [inline src/containers/User/ResetPassword/ResetForm.jsx]: Same.

File: src/containers/User/Signup/Form.jsx [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+        title={"Before you sign up"}
+        confirmMessage="I understand, sign me up"
+        onClose={onModalCancel}
+        onConfirm={onModalConfirm}
+      />
+      <FormWrapper
+        initialValues={initialValues}
+        loading={!validationSchema}
+        validationSchema={validationSchema}
+        onSubmit={async (...args) => {
```
Comment (tiagoalvesdulce) [inline src/containers/User/Signup/Form.jsx]: Same.

File: src/containers/User/VerifyEmail/RequestForm.jsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        confirmMessage="I understand, continue"
+        onClose={onModalCancel}
+        onConfirm={onModalConfirm}
+      />
+      <FormWrapper
+        initialValues={{
+          email: ""
+        }}
+        validationSchema={validationSchema}
+        onSubmit={async (...args) => {
```
Comment (tiagoalvesdulce) [inline src/containers/User/VerifyEmail/RequestForm.jsx]: Same.

File: src/containers/Header/hooks.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import * as sel from "src/selectors";
+import * as act from "src/actions";
+import { useRedux } from "src/redux";
+
+const mapStateToProps = {
```
Comment (tiagoalvesdulce) [inline src/containers/Header/hooks.js]: You should include the `loggedInAsUsername` selector here so the navbar keeps consistent with the username if it changes.

Comment (fernandoabolafio) [reply src/containers/Header/hooks.js]: Not sure I understand your comment. I am using the `loggedInAsUsername` selector

Comment (tiagoalvesdulce) [reply src/containers/Header/hooks.js]: I was in an outdated version. Sorry about that.

**Comment by tiagoalvesdulce** [comment]
Two minor things that I noticed:

1. The space between the `Already have an account?` and `Log in!` is too big. Should be equal the space between the `Dont have an account?` and `Create here!`. See prints below:

space

Screen Shot 2019-06-18 at 11 57 36

2. The hover effect on links is moving elements near to it. Also, it doesn't have a smooth transition. See:

bug1

**Comment by fernandoabolafio** [comment]
@tiagoalvesdulce fixed your comment about the span tags and the '1'. For 2 I created an issue in the pi-ui repo: https://github.com/tiagoalvesdulce/pi-ui/issues/56.
